src/rans/rANSCoder.py

```diff
@@ -50,10 +56,15 @@ def find_in_int_dist(cdf, to_find):
 
     print ("Error: Could not find symbol in integer-dist")
 
+spec = [
+    ('state', numba.int64),
+    ('encoded_data', types.ListType(types.int64)),
+]
+@numba.experimental.jitclass(spec=spec)
 class Encoder:
```

Comment:
Could you also do the same thing with numba for the `Decoder` class further down in the file?